### PR TITLE
Support Causal Convolution.

### PR DESCRIPTION
**Pull request by ds-hwang**
jax.lax.conv_general_dilated doesn’t natively support "CAUSAL" mode, so it implement causal padding convolution.

For example, with a window size of 3 and a stride of 2, padding="SAME" is applied as padding=((1,1),) as shown below, as keeping input and output length.
```
          pad|         |pad
paddings:   0|0 0 0 0 0|0
            |___|
                |___|
                    |___|
```

To implement padding="CAUSAL", we use padding=((2,0),) as follows.
```
           pad|         |pad
paddings:  0 0|0 0 0 0 0|
           |___|
               |___|
                   |___|
```

The first component is time and treated as "CAUSAL", while the remaining components are handled with "SAME" padding.

### CODE REVIEW

**Comment by ds-hwang** [comment]
@ruomingp, @markblee Could you take a look?

**Review by ruomingp — APPROVED** [review]

**Review by markblee — APPROVED** [review]

**Comment by ds-hwang** [comment]
Thank you for review!
